presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcPageSink.java

```diff
@@ -165,6 +169,22 @@ else if (DATE.equals(type)) {
             long localMillis = getInstanceUTC().getZone().getMillisKeepLocal(DateTimeZone.getDefault(), utcMillis);
             statement.setDate(parameter, new Date(localMillis));
         }
+        else if (type instanceof TimestampType) {
+            long timestampValue = type.getLong(block, position);
+            int nanos = ((TimestampType) type).getPrecision() == MILLISECONDS ?
```

Comment:
Just thinking would it make sense to move `get nanos` and `get millis` to `com.facebook.presto.common.type.TimestampType` although this would be the only place to use them.